hive-plugins/src/main/java/co/cask/plugin/etl/batch/commons/HiveSchemaConverter.java

```diff
@@ -49,12 +49,14 @@
    */
   public static HCatSchema toHiveSchema(Schema schema, HCatSchema tableSchema) {
     List<HCatFieldSchema> fields = Lists.newArrayList();
+    //TODO: Finalize a consistent way to whether do projection on source or not while reading and if needed
```

Comment:
With HCatalog to get  the schema of the table we need to first set the input by calling HCatInputFormat.setInput(Job, ....) This call sets properties in the hadoop configuration. Since, conf is not available during configurePipeline I don't think it will be possible.